scripts/load-test/collect-pod-logs/collect-openshift-pod-logs.sh

```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
+
+# use the set -e option to make the script exit immediately if a command fails.
+set -e
+
+# Array of default namespaces
+default_namespaces=(
+    "application-service"
+    "build-service"
+    "enterprise-contract-service"
+    "gitops-service-argocd"
+    "openshift-user-workload-monitoring"
+    "integration-service"
+    "internal-services"
+    "jvm-build-service" 
+    "openshift-service-ca"
+    "openshift-service-ca-operator"
+    "release-service")
+
+
+main() {
+    declare -g log_dir
+    # Log directory configuration
+    # Format:  ./collect-openshift-pod-logs.sh logs-${USER}-$(date +%Y-%m-%d)  
+
+    # if no parameter given, a default will be used for the log_dir
+    if [ -z "$1" ]; then
+      # Set default log directory
+      log_dir="logs-${USER}-$(date +%Y-%m-%d)"
+    else
+      # Use log directory provided as script argument
+      log_dir="$1"
+    fi
+
+    # Create log directory if it doesn't exist
+    mkdir -p "${log_dir}" || { echo "Error: Could not create directory ${log_dir}"; exit 1; }
+
+    # Read namespaces from a file if it exists, otherwise use the default namespaces array
+    namespaces_file="namespaces.txt"
+    if [ -f "${namespaces_file}" ]; then
+      mapfile -t namespaces < "${namespaces_file}"
+    else
+      namespaces=("${default_namespaces[@]}")
+    fi
+
+    # Trap SIGINT (Ctrl+C) and call cleanup_and_exit function
+    trap 'cleanup_and_exit' SIGINT
+
+    # Start collecting new tailed log data from (namespaces in namespace array) namespace pods new containers
+    collect_logs_from_existing_namespaces
+
+    # Start collecting new data from all new tenant type namespace pods containers
+    collect_logs_from_new_namespaces
+}
+
+# Function to tail logs of a created or existing pod containers in the background and store it in a distinct file
+collect_logs() {
+  local namespace=$1
+  local pod_name=$2
+  local container_name=$3
+  local container_dir=$4
+  local log_file="${container_dir}/${container_name}.log"
+
+  echo "Collecting logs for container ${container_name} in pod ${pod_name} in namespace ${namespace} ..."
+  kubectl logs -f "${pod_name}" -c "${container_name}"  -n "${namespace}" --tail=1 >> "${log_file}" 2>&1 &
```

Comment:
@naftalysh I have another observation from using the script along with the running load test.

When the pod/container is created and the script reacts to the event by starting to collect the log, sometimes the pod is not yet running and the collection of the logs fails like [in the following example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37577/rehearse-37577-periodic-ci-redhat-appstudio-e2e-tests-main-load-test/1641019187222548480/artifacts/load-test/redhat-appstudio-load-test/artifacts/loadtest-0001-tenant/loadtest-0001-component-4sql4-init-pod/step-init.log):
```
Error from server (BadRequest): container "step-init" in pod "loadtest-0001-component-4sql4-init-pod" is waiting to start: PodInitializing
```

So, the script should wait until the pod/container is actually running before it starts collecting the log for it.

Reply:
> @naftalysh I have another observation from using the script along with the running load test.
> 
> When the pod/container is created and the script reacts to the event by starting to collect the log, sometimes the pod is not yet running and the collection of the logs fails like [in the following example](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37577/rehearse-37577-periodic-ci-redhat-appstudio-e2e-tests-main-load-test/1641019187222548480/artifacts/load-test/redhat-appstudio-load-test/artifacts/loadtest-0001-tenant/loadtest-0001-component-4sql4-init-pod/step-init.log):
> 
> ```
> Error from server (BadRequest): container "step-init" in pod "loadtest-0001-component-4sql4-init-pod" is waiting to start: PodInitializing
> ```
> 
> So, the script should wait until the pod/container is actually running before it starts collecting the log for it.

Updated the collect-openshift-pod-logs.sh script by adding the following checks:
Per each pod - checks that its phase=Running
Per each container - checks that its ready status='true' and its status='running'